lib/pcb/svg-object-fns/create-svg-objects-for-soldermask-copper-pours.ts

```diff
@@ -0,0 +1,261 @@
+    }
+  }
+
+  // Add soldermask layer (excluding uncovered pours)
+  // Use darker green color for better contrast with covered copper pours
+  const darkerGreen = darkenColor(soldermaskColor)
+  if (!maskPolygon.isEmpty()) {
+    objects.push({
+      name: "path",
+      type: "element",
+      value: "",
+      children: [],
+      attributes: {
+        class: "pcb-soldermask",
+        d: polygonToPathD(maskPolygon),
+        fill: darkerGreen,
+        "fill-opacity": "0.85",
+        "data-type": "pcb_soldermask",
+        "data-pcb-layer": `${currentLayer}-soldermask`,
+      },
+    })
+  }
+
+  // Create brighter green layer for covered copper pours
+  for (const pour of coveredPours) {
+    const pourPolygon = createCopperPourPolygon(pour, ctx)
+    if (pourPolygon) {
+      try {
+        const coveredArea = BooleanOperations.intersect(
+          boardPolygon,
+          pourPolygon,
+        )
+        if (!coveredArea.isEmpty()) {
+          objects.push({
+            name: "path",
+            type: "element",
+            value: "",
+            children: [],
+            attributes: {
+              class: "pcb-soldermask-over-copper",
+              d: polygonToPathD(coveredArea),
+              fill: brighterGreen,
+              "fill-opacity": "0.9",
+              "data-type": "pcb_soldermask_over_copper",
+              "data-pcb-layer": `${currentLayer}-soldermask`,
+            },
+          })
+        }
+      } catch (e) {
+        console.warn("Failed to create soldermask over copper pour:", e)
+      }
+    }
+  }
+
+  return objects
+}
+
+function createBoardPolygon(
+  board: PCBBoard,
+  ctx: PcbContext,
+): Flatten.Polygon | null {
+  const { transform } = ctx
+  const { width, height, center, outline } = board
+
+  let points: [number, number][]
+
+  if (outline && Array.isArray(outline) && outline.length >= 3) {
+    points = outline.map((point) => {
+      const [x, y] = applyToPoint(transform, [point.x, point.y])
+      return [x, y] as [number, number]
+    })
+  } else if (
+    width &&
+    height &&
+    center.x !== undefined &&
+    center.y !== undefined
+  ) {
+    const halfWidth = width / 2
+    const halfHeight = height / 2
+
+    const corners = [
+      [center.x - halfWidth, center.y - halfHeight],
+      [center.x + halfWidth, center.y - halfHeight],
+      [center.x + halfWidth, center.y + halfHeight],
+      [center.x - halfWidth, center.y + halfHeight],
+    ]
+
+    points = corners.map((corner) => {
+      const [x, y] = corner
+      if (x === undefined || y === undefined) {
+        throw new Error("Invalid corner coordinates")
+      }
+      const result = applyToPoint(transform, [x, y]) as [number, number]
+      return [result[0], result[1]] as [number, number]
+    })
+  } else {
+    return null
+  }
+
+  try {
+    return new Polygon(points.map(([x, y]) => flattenPoint(x, y)))
+  } catch (e) {
+    console.warn("Failed to create board polygon:", e)
+    return null
+  }
```

Comment:
does this code exist in multiple files? please don't copy and paste code around (DRY)